src/main/java/de/tum/in/www1/artemis/service/VcsAccessLogService.java

```diff
@@ -0,0 +1,86 @@
+package de.tum.in.www1.artemis.service;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Repository;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.participation.Participation;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.vcstokens.AuthenticationMechanism;
+import de.tum.in.www1.artemis.domain.vcstokens.VcsAccessLog;
+import de.tum.in.www1.artemis.repository.ParticipationRepository;
+import de.tum.in.www1.artemis.repository.VcsAccessLogRepository;
+import de.tum.in.www1.artemis.web.rest.repository.RepositoryActionType;
+
+@Profile(PROFILE_CORE)
+@Service
+public class VcsAccessLogService {
+
+    private static final Logger log = LoggerFactory.getLogger(VcsAccessLogService.class);
+
+    private final VcsAccessLogRepository vcsAccessLogRepository;
+
+    private final ParticipationRepository participationRepository;
+
+    VcsAccessLogService(VcsAccessLogRepository vcsAccessLogRepository, ParticipationRepository participationRepository) {
+        this.vcsAccessLogRepository = vcsAccessLogRepository;
+        this.participationRepository = participationRepository;
+    }
+
+    /**
+     * Creates a vcs access log entry and stores it to the database
+     *
+     * @param user                    The user accessing the repository
+     * @param participation           The participation which owns the repository
+     * @param actionType              The action type: READ or WRITE
+     * @param authenticationMechanism The used authentication mechanism: password, vcs token (user/participation) or SSH
```

Comment:
As it's also possible to have `AuthenticationMechanism.CODE_EDITOR`, I think this should be added to the doc string too